src/Raven.Server/Documents/Handlers/Processors/Documents/AbstractDocumentHandlerProcessorForGet.cs

```diff
@@ -0,0 +1,451 @@
+
+                // includes
+                var counters = RequestHandler.GetStringValuesQueryString("counter", required: false);
+                var revisions = GetRevisionsToInclude();
+                var timeSeries = GetTimeSeriesToInclude();
+                var compareExchangeValues = RequestHandler.GetStringValuesQueryString("cmpxchg", required: false);
+
+                responseWriteStats = await GetDocumentsByIdAsync(context, ids, includePaths, revisions, counters, timeSeries, compareExchangeValues, metadataOnly, etag);
+            }
+            else
+            {
+                pageSize = RequestHandler.GetPageSize();
+                actionName = nameof(GetDocumentsAsync);
+
+                var changeVector = RequestHandler.GetStringFromHeaders(Constants.Headers.IfNoneMatch);
+                var etag = RequestHandler.GetLongQueryString("etag", false);
+
+                var isStartsWith = HttpContext.Request.Query.ContainsKey("startsWith");
+
+                StartsWithParams startsWithParams = null;
+
+                if (isStartsWith)
+                {
+                    startsWithParams = new StartsWithParams
+                    {
+                        IdPrefix = HttpContext.Request.Query["startsWith"],
+                        Matches = HttpContext.Request.Query["matches"],
+                        Exclude = HttpContext.Request.Query["exclude"],
+                        StartAfterId = HttpContext.Request.Query["startAfter"],
+                    };
+                }
+
+                responseWriteStats = await GetDocumentsAsync(context, etag, startsWithParams, metadataOnly, changeVector);
+            }
+
+            if (responseWriteStats != NoResults)
+            {
+                AddPagingPerformanceHint(PagingOperationType.Documents, actionName, HttpContext.Request.QueryString.Value, responseWriteStats.NumberOfResults,
+                    pageSize, sw.ElapsedMilliseconds, responseWriteStats.TotalDocumentsSizeInBytes);
+            }
+        }
+    }
+
+    protected async ValueTask<(long NumberOfResults, long TotalDocumentsSizeInBytes)> GetDocumentsByIdAsync(TOperationContext context, StringValues ids, StringValues includePaths, RevisionIncludeField revisions,
+        StringValues counters, HashSet<AbstractTimeSeriesRange> timeSeries, StringValues compareExchangeValues, bool metadataOnly, string etag)
+    {
+        var result = await GetDocumentsByIdImplAsync(context, ids, includePaths, revisions, counters, timeSeries, compareExchangeValues, metadataOnly, etag);
+
+        if (result.StatusCode == HttpStatusCode.NotFound)
+        {
+            if (etag == HttpCache.NotFoundResponse)
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+            else
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+
+            return NoResults;
+        }
+
+        if (etag == result.Etag)
```

Comment:
Can you calculate the etag properly without the missing includes?

Reply:
It wasn't working as expected. Now I get the etag of the missing includes into account as well,